src/back/auth/src/repo/table_session.cpp

```diff
@@ -89,15 +102,44 @@ void Session::UpdateTokens(const model::Session& s)
 }
 
 const storages::postgres::Query kMarkSessionInactive{
-	"UPDATE session SET active = false"
+	"UPDATE session SET active = false "
 	"WHERE id=$1",
 	storages::postgres::Query::Name{"mark_session_inactive"},
 };
 
+const storages::postgres::Query kMarkSessionInactiveAndCleanTokens{
+	"UPDATE session SET active=false, token='', accessToken='', refreshToken='', idToken='' "
+	"WHERE id=$1",
+	storages::postgres::Query::Name{"mark_session_inactive_and_clean_tokens"},
+};
+
 void Session::MarkInactive(const model::Session& s)
 {
 	_pg->Execute(
 		storages::postgres::ClusterHostType::kMaster, kMarkSessionInactive, s._id);
 }
 
+void Session::MarkInactiveAndCleanTokensById(const boost::uuids::uuid& sid)
+{
+	_pg->Execute(
+		storages::postgres::ClusterHostType::kMaster, kMarkSessionInactiveAndCleanTokens, sid);
+}
+
+void Session::Refresh(
+	const model::Session& data,
+	const boost::uuids::uuid& oldSessionId)
+{
+	storages::postgres::Transaction t =
+		_pg->Begin("refresh_session_transaction",
+			storages::postgres::ClusterHostType::kMaster, {});
+
+	std::apply([&t](const auto&... args) {
+		t.Execute(kInsertSession, args...);
+	}, boost::pfr::structure_tie(data));
+
+	MarkInactiveAndCleanTokensById(oldSessionId);
```

Comment:
Тут запрос не в транзакции.
```suggestion
	t.Execute(kMarkSessionInactive, oldId);
```

Reply:
Поправил.